### PR TITLE
Add MSI installer and set icon

### PR DESCRIPTION
**Pull request by hombit**
Add icon into .exe file
Create MSI in Appveyor and save it as artifact
Set minimum CMake version to 3.0.0

### CODE REVIEW

File: dist/windows/WIX.template.in [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        <?endif?>
+
+        <?ifdef CPACK_WIX_UI_DIALOG?>
+        <WixVariable Id="WixUIDialogBmp" Value="$(var.CPACK_WIX_UI_DIALOG)"/>
+        <?endif?>
+
+        <FeatureRef Id="ProductFeature"/>
+
+        <DirectoryRef Id="TARGETDIR">
+            <Merge Id="VCRedist" SourceFile="C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Redist\MSVC\14.10.25008\MergeModules\Microsoft_VC150_CRT_x64.msm" DiskId="1" Language="0"/>
```
Comment (matwey) [inline dist/windows/WIX.template.in]: Are you sure that this is going to work in Appveyor where we use Studio 2015?

Comment (hombit) [inline dist/windows/WIX.template.in]: This doesn't work even with my Studio 2017. This file isn't used, I keep it just as an example for future.

Comment (ForNeVeR) [inline dist/windows/WIX.template.in]: FYI: we can't use VC++'s merge modules if we want to target outdated Windows 7 machines.

New (VS2015 and VS2017) VC++ redistributable's merge module depends on some WinRT API (sic!). This API should normally be available with automatic updates, but some Windows 7 machines never received it, so they have no `api-ms-win-crt-runtime-l1-1-0.dll` installed.

[See here](https://stackoverflow.com/questions/35072133/visual-studio-2015-c-app-requires-api-ms-win-crt-runtime-l1-1-0-dll-on-client) for more information.

VC2015/VC2017 merge modules doesn't contain the necessary DLLs, while the `vsredist_{x86,x64}.exe` installs them properly. That's why we cannot have nice things (merge modules) in the current situation.

Comment (matwey) [inline dist/windows/WIX.template.in]: @ForNeVeR do you know KB-numbers for updates bringing ```api-ms-*.dll```?

Comment (matwey) [inline dist/windows/WIX.template.in]: I see ```KB2999226```

**Pull request closed without merging**